odoo/addons/base/models/res_config.py

```diff
@@ -316,12 +303,24 @@ def execute(self):
         _logger.info('Selecting addons %s to install', to_install)
 
         IrModule = self.env['ir.module.module']
-        modules = []
+        modules = self.env['ir.module.module']
+        to_install_missing_names = []
         for name in to_install:
             module = IrModule.search([('name', '=', name)], limit=1)
-            modules.append((name, module))
+            if module:
+                modules += module
+            else:
+                to_install_missing_names.append(name)
```

Comment:
Could be simpler, and with only one search:
```python
modules = IrModule.search([('name', 'in', to_install)])
module_names = {module.name for module in modules}
to_install_missing_names = [name for name in to_install if name not in module_names]
```
I don't think that the order of `modules` is relevant, so this might be fine.